### PR TITLE
Define size_t and ptrdiff_t

### LINKED ISSUE #376: Should calling convention also define ptrdiff_t?
[cc-type-sizes-and-alignments](https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-cc.adoc#cc-type-sizes-and-alignments) has the definitions of types for both IPL32 and LP64 systems. Would it help to define the size of `ptrdiff_t`?

### PR DESCRIPTION
**Pull request by MaskRay**
Close #376

### CODE REVIEW

File: riscv-cc.adoc [hunk 1]
```diff
@@ -426,6 +426,10 @@ A null pointer (for all types) has the value zero.
 corresponding real type (`float`, `double`, or `long double`), with the first
 member holding the real part and the second member holding the imaginary part.
 
+The type `size_t` is defined as `unsigned int` for RV32 and `unsigned long` for RV64.
+
+The type `ptrdiff_t` is defined as `int` for RV32 and `long` for RV64.
```
Comment (appujee) [inline riscv-cc.adoc:431]: Should we put them in table above as well? I feel like it is easier to follow in a table.

Comment (jrtc27) [reply riscv-cc.adoc:431]: They're not basic types, so no

**Review by appujee — APPROVED** [review]
LTGM, but i'd let other reviewers take the final call.

**Review by kito-cheng — APPROVED** [review]
LGTM, this is document what we have, checked it consistent with clang and GCC.

**Comment by kito-cheng** [comment]
@appujee @MaskRay thanks!

**Pull request merged**